packages/js/test/unit/server/integrations/uncaught_exception.server.test.ts

```diff
@@ -0,0 +1,122 @@
+import defaultExport, { exportedForTesting } from '../../../../src/server/integrations/uncaught_exception'
+import { TestTransport, TestClient, nullLogger } from '../../helpers'
+import * as util from '../../../../src/server/util'
+import * as aws from '../../../../src/server/aws_lambda'
+import Singleton from '../../../../src/server'
+const { 
+  handleUncaughtException, 
+  setIsReporting,
+  setHandlerAlreadyCalled,
+} = exportedForTesting
+
+describe('Uncaught Exception', () => {
+  let client: typeof Singleton
+  let fatallyLogAndExitSpy: jest.SpyInstance
+  let notifySpy: jest.SpyInstance
+
+  beforeEach(() => {
+    // We just need a really basic client, so ignoring type issues here
+    client = new TestClient(
+      { apiKey: 'testKey', afterUncaught: jest.fn(), logger: nullLogger() }, 
+      new TestTransport()
+    ) as unknown as typeof Singleton
+    // Have to mock fatallyLogAndExit or we will crash the test
+    fatallyLogAndExitSpy = jest
+      .spyOn(util, 'fatallyLogAndExit')
+      .mockImplementation(() => true as never)
+    notifySpy = jest.spyOn(client, 'notify')
+  })
+
+  afterEach(() => {
+    jest.clearAllMocks()
+    process.removeAllListeners('uncaughtException')
+    setIsReporting(false)
+    setHandlerAlreadyCalled(false)
+  })
+
+  describe('default export plugin', () => {
+    it('is a function which returns an object with a load function', () => {
+      expect(defaultExport()).toStrictEqual({
+        load: expect.any(Function)
+      })
+    })
+
+    describe('load', () => {
+      const load = defaultExport().load
+
+      it('adds a listener for uncaughtException', () => {
+        load(client)
+        const listeners = process.listeners('uncaughtException')
+        expect(listeners.length).toBe(1)
+        expect(listeners[0].name).toBe('honeybadgerUncaughtExceptionListener') 
+
+        const error = new Error('uncaught')
+        process.emit('uncaughtException', error)
+        expect(notifySpy).toHaveBeenCalledTimes(1)
+      })
+
+      it('removes the AWS lambda uncaught exception listener', () => {
+        const restoreEnv = process.env
```

Comment:
Does this give you a copy of `process.env`? Or is it a reference?

Reply:
Ah good catch, I forgot the braces! Added.